securesystemslib/key.py

```diff
@@ -63,16 +60,16 @@ class SSlibKey(Key):
 
     def __init__(
         self,
-        keyid: Optional[str],
         keytype: str,
         scheme: str,
         keyval: Dict[str, str],
+        keyid: Optional[str] = None,
         unrecognized_fields: Optional[Dict[str, Any]] = None,
     ):
-        self.keyid = keyid if keyid else None
         self.keytype = keytype
         self.scheme = scheme
         self.keyval = keyval
+        self.keyid = keyid if keyid else None
```

Comment:
You no longer need to make checks about what is inside `keyid`.
```suggestion
        self.keyid = keyid
```

Reply:
@PradyumnaKrishna I think this comment still applies as you can remove this part as you have the default value.

Reply:
I converted a empty string to `None` here, which can be present in securesystemslib key `dict`.

```python3
>>> keyid = ""
>>> keyid = keyid if keyid else None
>>> print(keyid)
None
```